lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/NetconfNmdaBaseServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+                .withChild(editContent).build();
+
+        Preconditions.checkNotNull(editStructure);
+
+        return domRpcService.invokeRpc(toPath(NETCONF_EDIT_DATA_QNAME),
+                NetconfMessageTransformUtil.wrap(NETCONF_EDIT_DATA_QNAME,
+                        getDatastoreNode(targetDatastore), getDefaultOperationNode(dataModifyActionAttribute.get()), editStructure));
+    }
+
+    private DataContainerChild<?, ?> getDatastoreNode(QName datastore) {
+        return Builders.leafBuilder().withNodeIdentifier(toId(QName.create(NETCONF_NMDA_EXTENSION_QNAME, "datastore")))
+                .withValue(datastore).build();
+    }
+
+    private DataContainerChild<?, ?> getConfigFilterNode(Boolean configFilter) {
+        return Builders.leafBuilder().withNodeIdentifier(toId(QName.create(NETCONF_NMDA_EXTENSION_QNAME, "config-filter")))
+                .withValue(configFilter).build();
+    }
+
+    private DataContainerChild<?, ?> getMaxDepthNode(Integer maxDepth) {
+        return Builders.leafBuilder().withNodeIdentifier(toId(QName.create(NETCONF_NMDA_EXTENSION_QNAME, "max-depth")))
+                .withValue(maxDepth).build();
+    }
+
+    private DataContainerChild<?, ?> getOriginFilterNode(Set<QName> originFilter) {
+        List<LeafSetEntryNode<Object>> leafSetEntryNodes = new ArrayList<>();
+        originFilter.forEach(originFilterEntry -> {
+            leafSetEntryNodes.add(Builders.leafSetEntryBuilder()
+                    .withNodeIdentifier(new YangInstanceIdentifier
+                            .NodeWithValue(QName.create(NETCONF_NMDA_EXTENSION_QNAME, "origin-filter"), originFilterEntry))
+                    .withValue(originFilterEntry)
+                    .build());
+        });
+        return Builders.leafSetBuilder().withNodeIdentifier(toId(QName.create(NETCONF_NMDA_EXTENSION_QNAME, "origin-filter")))
+                .withValue(leafSetEntryNodes).build();
+    }
+
+    private DataContainerChild<?, ?> getNegatedOriginFilterNode(Set<QName> negatedOriginFilter) {
+        List<LeafSetEntryNode<Object>> leafSetEntryNodes = new ArrayList<>();
+        negatedOriginFilter.forEach(negatedOriginFilterEntry -> {
+            leafSetEntryNodes.add(Builders.leafSetEntryBuilder()
+                    .withNodeIdentifier(new YangInstanceIdentifier
+                            .NodeWithValue(QName.create(NETCONF_NMDA_EXTENSION_QNAME, "negated-origin-filter"), negatedOriginFilterEntry))
+                    .withValue(negatedOriginFilterEntry)
+                    .build());
+        });
+        return Builders.leafSetBuilder().withNodeIdentifier(toId(QName.create(NETCONF_NMDA_EXTENSION_QNAME, "negated-origin-filter")))
+                .withValue(leafSetEntryNodes).build();
+    }
+
+    private DataContainerChild<?, ?> getWithOriginNode() {
+        return Builders.leafBuilder().withNodeIdentifier(toId(QName.create(NETCONF_NMDA_EXTENSION_QNAME, "with-origin")))
```

Comment:
These QNames should have corresponding NodeIdentifier constants created through NodeIdentifier.create()

Reply:
Please, see the new commit.